ebean-core/src/main/java/io/ebeaninternal/server/querydefn/DefaultOrmQuery.java

```diff
@@ -1447,6 +1447,21 @@ public final Query<T> fetchLazy(String path, String properties) {
   }
 
   private Query<T> fetchInternal(String path, String properties, FetchConfig config) {
+    // look at all One Props, see if we can select only the ID of one?
+    for (BeanPropertyAssocOne<?> beanPropertyAssocOne : beanDescriptor.propertiesOne()) {
+      if(beanPropertyAssocOne.name().equals(path)) {
+        // only if we select the ID do we prevent the join, otherwise we need something from the other table anyway...
+        if(beanPropertyAssocOne.targetIdProperty().equals(properties)) {
```

Comment:
this if branch does not work properly yet, you can also fetch other OnProperty, not just the id